sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -363,11 +358,36 @@ BlobContainerClient getContainerClient(BlobClient client) {
     }
 
     /**
+     * As noted by the NIO docs, this method is not atomic. It is possible to delete a file in use by another process,
+     * and doing so will not immediately invalidate any channels open to that file--they will simply start to fail.
+     * Root directories cannot be deleted even when empty.
```

Comment:
I don't think we need to call out what the NIO documents state explicitly, let's just include it as a statement in our docs. Would also like to change the first line to call out what exactly this method does, such as `This API deletes the specified path/file`, then adding the ancillary info like this isn't atomic, what can't be deleted, etc.

```suggestion
     * Deletes the specified path.
     * <p>
     * This method is not atomic, it is possible to delete a file in use by another process. Open channels to the file won't immediately invalidate, they will simply start to fail.
     * <p>
     * Root directories cannot be deleted even when empty.
```

Reply:
If we really want to call out the NIO doc, we could also just link it

Reply:
Below the javadocs I've written, I always inherit the javadocs from the parent, so all the extra info should be there anyway. I'll update the docs to remove the NIO reference and change the first sentence.